### ISSUE TITLE
Remove [SetUp] Tag from Test Data Initialization in PackageHelpersTest

### ISSUE THREAD

**Issue by Tohron** [dotnet/arcade#6901]
Chad raised concerns about the use of the [SetUp] tag in
https://dev.azure.com/dnceng/internal/_git/dotnet-release/pullrequest/12458?_a=files&discussionId=59293&path=%2Fsrc%2FMicrosoft.DotNet.Release%2FMicrosoft.DotNet.ReleaseLib.Tests%2FPackageHelpersTests.cs

This issue is for replacing that usage with a better implementation.

**Comment by adiaaida** [comment]
In case the link to the PR stops working, here is the discussion:

> Missy Messa
> Monday
> 
> Resolved
> Word of caution: the [SetUp] attribute means that this will run for EVERY test that you have in your file. If there are tests that don't need this data to be set up, it will be wasted time spent for that test. I would strongly suggest only running InitializeTestData on the tests that absolutely need it. Also, it would be even better if InitiallizeTestData returned data instead of setting class-wide variables, that are now stateful.
> 
> Chad Nedzlek
> Monday
> I'm shocked these tests pass as is. They are likely only passing because the tests in this assembly aren't running in parallel, which is a dangerous thing for a test to depend on.
> 
> Michelle McDaniel
> 5h ago
> Just a note: there ARE tests that use the test data. Without this function marked as [SetUp], the test data is never being setup, because InitializeTestData is never called. You need to work with @🔥 Matt Galbraith 🔥, who wrote this up initially, to determine what tests are using what data, so they can still get their test data.
> 
> Missy Messa
> 4h ago
> Note: Michelle and I are not giving you conflicting information. The SetUp attribute says "this method will run before each test that is called in this class". If you remove the attribute (as you did), you now need to explicitly call the method in the tests that require that test data, as I mentioned in my initial comment, and as Michelle is pointing out as well, that there are tests that require that test data to be initialized.
> 
> I also mentioned that instead of setting class-wide variables to be used in the tests (thus, my comments about state, and Chad's comment above about how this is a dangerous scenario for test data), the InitializeTestData method should return actual values to assign to locally-scoped variables and not merely set class-wide variables.
> 
> Missy Messa
> 4h ago
> Also, please get into the habit of running your test suites locally before pushing up changes so you can ensure that your tests pass. It will avoid unnecessary churn on your PR.
> 
> 🔥 Matt Galbraith 🔥
> 2h ago
> 
> From today's standup discussion, Aaron is OK to make a follow-up issue to do this part and just assign it to me, will deal with it after.
